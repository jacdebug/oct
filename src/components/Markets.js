import React from 'react';
import styled from 'styled-components';
import MarketsNav from './MarketsNav';
import { Switch, Route } from 'react-router-dom';
import MarketSelectedContainer from '../containers/MarketSelectedContainer';

const Markets = ({ ...props }) =>
  <MainContentWrapper>
    <MarketsNav {...props} />
    <Switch>
      {props.market.marketsLoaded.map((market, index) =>
        <Route
          key={index}
          path="/:pageid?"
          component={MarketSelectedContainer}
        />
      )}
    </Switch>
  </MainContentWrapper>;

export default Markets;

const MainContentWrapper = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  min-height: 500px;
  margin: 0 40px;
`;
