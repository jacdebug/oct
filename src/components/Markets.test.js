import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Markets from './Markets';

const market = { marketsLoaded: [] };

it('renders without crashing', () => {
  shallow(<Markets market={market} />);
});

it('renders a snapshot', () => {
  const tree = shallow(<Markets market={market} />);
  expect(toJson(tree)).toMatchSnapshot();
});
