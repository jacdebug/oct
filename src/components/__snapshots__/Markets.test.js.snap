// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<styled.div>
  <MarketsNav
    market={
      Object {
        "marketsLoaded": Array [],
      }
    }
  />
  <Switch />
</styled.div>
`;
